crates/oxc_semantic/src/lib.rs

```diff
@@ -187,4 +187,55 @@ mod tests {
             }
         }
     }
+
+    #[test]
+    fn test_reference_resolutions_simple_read_write() {
+        let alloc = Allocator::default();
+        let target_symbol_name = Atom::from("a");
+        let sources = [
+            ("let a = 1, b; b = ++a", ReferenceFlag::read_write()),
+            // parens are pass-through
+            ("let a = 1, b; b = (a)", ReferenceFlag::read()),
+            ("let a = 1, b; b = ++(a)", ReferenceFlag::read_write()),
+            ("let a = 1, b; b = ++((((a))))", ReferenceFlag::read_write()),
+            // simple binops/calls for sanity check
+            ("let a, b; a + b", ReferenceFlag::read()),
+            ("let a, b; b(a)", ReferenceFlag::read()),
+            ("let a, b; a = 5", ReferenceFlag::write()),
+            // unary op counts as write, but checking continues up tree
+            ("let a = 1, b; b = ++a", ReferenceFlag::read_write()),
+            ("let a = 1, b; b = --a", ReferenceFlag::read_write()),
+            ("let a = 1, b; b = a++", ReferenceFlag::read_write()),
+            ("let a = 1, b; b = a--", ReferenceFlag::read_write()),
+            // assignment expressions count as read-write
+            ("let a = 1, b; b = a += 5", ReferenceFlag::read_write()),
+            ("let a = 1; a += 5", ReferenceFlag::read_write()),
+            ("let a, b; b = a = 1", ReferenceFlag::read_write()),
+            ("let a, b; b = (a = 1)", ReferenceFlag::read_write()),
+            // sequences return last value in sequence
+            ("let a, b; b = (0, a++)", ReferenceFlag::read_write()),
+            // identifiers not in last value are also considered a read (at
+            // least, or now)
+            ("let a, b; b = (a, 0)", ReferenceFlag::read()),
+            ("let a, b; b = (--a, 0)", ReferenceFlag::read_write()),
+        ];
+
+        for (source, flag) in sources {
+            let semantic = get_semantic(&alloc, source, SourceType::default());
+            let a_id =
+                semantic.scopes().get_root_binding(&target_symbol_name).unwrap_or_else(|| {
+                    panic!("no references for '{target_symbol_name}' found");
+                });
+            let a_refs: Vec<_> = semantic.symbol_references(a_id).collect();
+            let num_refs = a_refs.len();
+
+            assert!(num_refs == 1, "expected to find 1 reference to '{target_symbol_name}' but {num_refs} were found\n\nsource:\n{source}");
+            if flag.is_write() {
```

Comment:
shouldn't these next two if statements just be `assert_eq!(a_refs[0], flag)`

Reply:
No, because a flag could be write or read/write